src/Voron/Impl/FileHeaders/HeaderAccessor.cs

```diff
@@ -101,6 +96,28 @@ public bool Initialize()
                 }
                 _revision = _theHeader->HeaderRevision;
 
+                if (_theHeader->Version != Constants.CurrentVersion)
+                {
+                    _locker.ExitWriteLock();
```

Comment:
Why are you exiting the write lock here?

Reply:
We're already under write lock so we release it so the updater can access the header accessor. We're in the constructor of storage env so nobody else can touch it anyway.